Hw 1

Varfalamei opened the pull request:
Настроили проект
Дописали тесты на модели
Сделали авторизацию по API_TOKEN

Varfalamei committed:
init HW_1

Mostovik71 committed:
Update views.py

Mostovik71 committed:
Update views.py

Mostovik71 committed:
Update views.py

Varfalamei committed:
update HW_1

Varfalamei committed:
update HW_1

Varfalamei committed:
update HW_1

Varfalamei committed:
update HW_1

Varfalamei committed:
add config

Varfalamei committed:
Merge branch 'dev' into HW_1

Varfalamei committed:
first attempt to pass tests

Varfalamei committed:
Merge remote-tracking branch 'origin/HW_1' into HW_1

Mostovik71 committed:
Update views.py

Mostovik71 committed:
Update views.py

Mostovik71 committed:
Update views.py

Varfalamei committed:
BugFix

Varfalamei committed:
Reformat code by linters

Varfalamei committed:
Reformat code by linters

Varfalamei committed:
Refactor tests

Varfalamei committed:
Refactor tests

ElisovaIra commented on pyproject.toml:
```diff
@@ -1,9 +1,9 @@
-authors = ["Emiliy Feldman <feldlime@yandex.ru>"]
-maintainers = ["Emiliy Feldman <feldlime@yandex.ru>"]
+authors = ["Shakirov Renat <renatshk@gmail.com>"]
+maintainers = ["Shakirov Renat <renatshk@gmail.com>"]
```
кажется где-то здесь должен быть список всей команды

Varfalamei replied:
Добавил

ElisovaIra commented on service/api/views.py:
```diff
@@ -1,50 +1,87 @@
+    if user_id > 10 ** 9:
         raise UserNotFoundError(error_message=f"User {user_id} not found")
 
+    if model_name != "model_1":
```
Если сервис будет использоваться в дальнейшем, то стоило бы заменить на `if model_name not in models`,  где `models` - список допустимых названий

Varfalamei replied:
Добавил библу моделей

ElisovaIra commented:
За исключением пары комментариев, хороший пулл реквест 💪
Понравилось, что есть описание и релевантное название ветки  

Добавьте еще тест на авторизацию, это +1 балл

ElisovaIra commented on tests/api/test_views.py:
```diff
@@ -33,9 +35,21 @@ def test_get_reco_success(
+    with client:
+        response = client.get(path, headers=headers)
+    assert response.status_code == HTTPStatus.NOT_FOUND
+    assert response.json()["errors"][0]["error_key"] == "model_not_found"
```
В ДЗ еще есть требование добавить тест на неуспешную авторизацию

Varfalamei replied:
Добавил

Varfalamei committed:
Добавил остальных авторов
Добавил базовый класс для будущих моделей
Добавил модель, которая выдает range(10) как реки
Добавил тест на авторизацию

Varfalamei committed:
Code refactoring (by linters)

Varfalamei committed:
Update models_zoo.py remove init from DumpModel

Varfalamei committed:
Add mypy config, clean views from unnecessary pydantic models, refactoring code by mypy
